src/App.tsx

```diff
@@ -1,14 +1,84 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import {
+  FILTER_FIELD_ACTIVE,
+  FILTER_FIELD_ALL,
+  FILTER_FIELD_COMPLETED,
+} from './tools/constants';
+
+// const prepareTodos = (todos: Todo[], title: string, filterField: string) => {
+//   const readyTodos = [...todos];
+
+//   if (title !== '') {
+//     return readyTodos.filter((todo: Todo) =>
+//       todo.title.toLowerCase().includes(title.toLowerCase()),
+//     );
+//   }
+
+//   if (filterField) {
+//     switch (filterField) {
+//       case FILTER_FIELD_ALL:
+//         return readyTodos;
+
+//       case FILTER_FIELD_ACTIVE:
+//         return readyTodos.filter(todo => !todo.completed);
+
+//       case FILTER_FIELD_COMPLETED:
+//         return readyTodos.filter(todo => todo.completed);
+
+//       default:
+//         return readyTodos;
+//     }
+//   }
+
+//   return readyTodos;
+// };
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [filterField, setFilterField] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(readyTodos => {
+        return readyTodos.filter((todo: Todo) =>
+          todo.title.toLowerCase().includes(title.toLowerCase()),
+        );
+      })
+      .then(readyTodos => {
+        switch (filterField) {
+          case FILTER_FIELD_ALL:
+            return readyTodos;
+
+          case FILTER_FIELD_ACTIVE:
+            return readyTodos.filter(todo => !todo.completed);
+
+          case FILTER_FIELD_COMPLETED:
+            return readyTodos.filter(todo => todo.completed);
+
+          default:
+            return readyTodos;
+        }
```

Comment:
Both filters can be merged into one and it would be better if you make helper like prepareTodos which will be defined in separate file and be called here.

Reply:
```suggestion
      .then(readyTodos => {
        return readyTodos.filter(todo => {
          const titleCondition = todo.title
            .toLowerCase()
            .includes(title.toLowerCase());

          switch (filterType) {
            case FilterType.All:
              return firstCondition;

            case  FilterType.Active:
              return firstCondition && !todo.completed;

            case FilterType.Completed:
              return firstCondition && todo.completed;

            default:
              return true;
          }
        });
      })
```

Reply:
This is the example of such function but as I mentioned before you have to create helper function for it and just call it when you receive the data.